tasks/util.py

```diff
@@ -1634,6 +1636,24 @@ def create_geom_summaries(self, output):
         generate_tile_summary(current_session(),
                               output._id, colid, output.table, colname)
 
+    def check_null_columns(self):
+        session = current_session()
+        result = session.execute('SELECT {values} FROM {table}'.format(
+                            table=self.output().table,
+                            values=', '.join(['sum(case when {column} is null then 0 else 1 end) {column}'.format(
+                                column=x) for x, _ in self.columns().iteritems()])))
```

Comment:
Since you went for analysing the table, you can do this very quickly with `pg_stats`

`SELECT attname FROM pg_stats WHERE schemaname = 'observatory' AND tablename = '{table}' AND null_frac = 1;`